src/api/java/com/minecolonies/api/compatibility/CompatibilityManager.java

```diff
@@ -653,11 +654,16 @@ private void discoverAllItems(final Level level)
                 {
                     stacks = tab.getDisplayItems();
                 }
+
+                final Object2IntLinkedOpenHashMap<Item> mapping = new Object2IntLinkedOpenHashMap<>();
                 for (final ItemStack item : stacks)
                 {
-                    setBuilder.add(new ItemStorage(item, true));
-                    listBuilder.add(item);
+                    if (mapping.addTo(item.getItem(), 1) > 100)
+                    {
+                        continue;
+                    }
```

Comment:
I would even go further here and add two tags.
First: Indexing blacklist. Which just skips the item entirely.
Second: Indexing whitelist. Which adds all sub-items, regardless of how many sub-items there are (skipping this check basically).

Additionally I would make that 100 configurable to the end user.

Reply:
Can go further, sure if anyone wants. But I dont see any immediate need for this.